### PR TITLE
Add edgemesh/edgemesh-gateway to addons

### PR DESCRIPTION
**Pull request by zhu733756**
Signed-off-by: zhu733756 <zhu733756@kubesphere.io>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Install edgemesh through keadm.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

/cc @fisherxu @zc2638 @Poorunga

### CODE REVIEW

**Review by gy95 — COMMENTED** [review]
delete tar file?: manifests/addons/edgemesh/edgemesh.tgz

**Comment by zhu733756** [comment]
> delete tar file?: manifests/addons/edgemesh/edgemesh.tgz

The last commit can help us to achieve this:
We can download the online chart tarball to the builtin dir or the local path.
- builtin dir: addons/edgemesh/edgemesh.tgz, this method will compile into the binary.
- local path: /tmp/addons/edgemesh/edgemesh.tgz, this method will download to the host path.

File: keadm/cmd/keadm/app/cmd/helm/installer.go [hunk 1]
```diff
@@ -83,6 +87,17 @@ func (cu *KubeCloudHelmInstTool) InstallTools() error {
 		baseHelmRoot = cu.ExternalHelmRoot
 	}
 
+	// Gets the profile key
+	if cu.Profile == "" {
+		cu.Profile = fmt.Sprintf("%s=v%s", VersionProfileKey, cu.ToolVersion.String())
+	}
+	// profile must be invalid
```
Comment (wackxu) [inline keadm/cmd/keadm/app/cmd/helm/installer.go:94]: profile must be valid?

Comment (wackxu) [reply keadm/cmd/keadm/app/cmd/helm/installer.go:94]: we shoud add validation for Profile

File: manifests/profiles/edgemesh-gateway.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    enable: true
+    listenPort: 20008
+    security:
+      enable: false
+      httpServer: ""
+      tlsCaFile: /etc/kubeedge/edgemesh/gateway/acls/rootCA.crt
+      tlsCertFile: /etc/kubeedge/edgemesh/gateway/acls/server.crt
+      tlsPrivateKeyFile: /etc/kubeedge/edgemesh/gateway/acls/server.key
+    enableHolePunch: true
+    transport: tcp
```
Comment (fisherxu) [inline manifests/profiles/edgemesh-gateway.yaml:65]: Add empty line here.

File: manifests/profiles/edgemesh.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      enable: true
+      listenPort: 20006
+      security:
+        enable: false
+        httpServer: ""
+        tlsCaFile: /etc/kubeedge/edgemesh/agent/acls/rootCA.crt
+        tlsCertFile: /etc/kubeedge/edgemesh/agent/acls/server.crt
+        tlsPrivateKeyFile: /etc/kubeedge/edgemesh/agent/acls/server.key
+      enableHolePunch: true
+      transport: tcp
```
Comment (fisherxu) [inline manifests/profiles/edgemesh.yaml:104]: ditto

File: manifests/addons/edgemesh/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# EdgeMesh Application
+
+Visit https://github.com/kubeedge/edgemesh/build/helm/edgemesh  for more information.
```
Comment (fisherxu) [inline manifests/addons/edgemesh/README.md:3]: Invalid url :)

File: manifests/addons/edgemesh-gateway/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# EdgeMesh Application
+
+Visit https://github.com/kubeedge/edgemesh/build/helm/edgemesh-gateway for more information.
```
Comment (fisherxu) [inline manifests/addons/edgemesh-gateway/README.md:3]: ditto

File: keadm/cmd/keadm/app/cmd/beta/init.go [hunk 6]
```diff
@@ -105,6 +106,9 @@ func addInitBetaJoinOtherFlags(cmd *cobra.Command, initBetaOpts *types.InitBetaO
 
 	cmd.Flags().StringVar(&initBetaOpts.ExternalHelmRoot, types.ExternalHelmRoot, initBetaOpts.ExternalHelmRoot,
 		"Add external helm root path to keadm.")
+
+	cmd.Flags().StringVar(&initBetaOpts.TarballURL, types.TarballURL, initBetaOpts.TarballURL,
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/beta/init.go:110]: So we must add the chart to release, right? :)

Comment (zhu733756) [reply keadm/cmd/keadm/app/cmd/beta/init.go:110]: Yes, edgemesh supports the tgz file in its codes, this way is also ok.

File: keadm/cmd/keadm/app/cmd/helm/renderer.go [hunk 7]
```diff
@@ -29,6 +35,9 @@ const (
 
 	// Chart Release Name
 	ReleaseName = "kubeedge"
+
+	// Default dir root for online chart tarball
+	DefaultDestDirRoot = ""
```
[the remark this answers is not in the document]
Comment (zhu733756) [reply keadm/cmd/keadm/app/cmd/helm/renderer.go:40]: Thanks, i will remove it.

**Pull request closed without merging**